libs/s25main/desktops/dskGameLobby.cpp

```diff
@@ -594,6 +614,10 @@ void dskGameLobby::Msg_Group_ComboSelectItem(const unsigned group_id, const unsi
 
 void dskGameLobby::GoBack()
 {
+    // We have to clean up the configured ai players, so that subsequent games can
+    // start with the default setup again.
+    GAMECLIENT.ClearAIBattlePlayers();
```

Comment:
I still think that `GameClient::Stop` is the best place to do this as here it can be easily forgotten at some point. I don't even know if `GameClient::ExitGame()` is the only place how a game is exited past this point, but `GameClient::Stop` is called in `GameClient::Start` (which would also be a good place: In `Start` it makes sense to clear AI players, but as it calls `Stop` already that might be even better. Maybe even an assert in `Start` that no AI battle is active could be added in case `Stop` is removed in the future)

Yes it's a bit of a mess ;-)

Reply:
Could it be you mean `GameClient::Connect` instead of `GameClient::Start`? `GameClient::Stop` is called by `GameClient::Connect`. I added an assert there.

Reply:
Yes indeed. Perfect!